Add "fly-speed" flag.

eth-p opened the pull request:
Similar to walk-speed, but for players who are flying.

Just a small change that is convenient for creative servers with large spawn areas.

aromaa commented on WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java:
```diff
@@ -0,0 +1,93 @@
+	@Override
+    public void initialize(Player player, Location current, ApplicableRegionSet set)
+	{
+		Double speed = WorldGuardUtils.queryValue(player, current.getWorld(), set.getRegions(), Flags.WALK_SPEED);
```
FLY_SPEED

eth-p replied:
Thanks... Copy paste blindness, sorry.

aromaa commented on WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java:
```diff
@@ -0,0 +1,93 @@
+	@Override
+	public boolean onCrossBoundary(Player player, Location from, Location to, ApplicableRegionSet toSet, Set<ProtectedRegion> entered, Set<ProtectedRegion> exited, MoveType moveType)
+	{
+		Double speed = WorldGuardUtils.queryValue(player, to.getWorld(), toSet.getRegions(), Flags.WALK_SPEED);
```
FLY_SPEED

aromaa commented on WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/wrappers/WorldGuardCommunicator.java:
```diff
@@ -1,5 +1,6 @@
 package net.goldtreeservers.worldguardextraflags.wg.wrappers;
 
+import net.goldtreeservers.worldguardextraflags.wg.handlers.*;
```
I don't really like to use the wildcard

eth-p replied:
My bad. My IDE automatically collapses multiple imports into wildcards, and I didn't have it turned off when opening the WGEF project. I'll fix that in the next commit.

aromaa reviewed (changes requested):
Theres also few formatting errors. Would be nice to combine walk speed and fly speed to abstract class as it has the same logic.

eth-p commented:
> Theres also few formatting errors.

Will fix that, sorry. I generally try to mimic the projecy style, but I miss space/tab differences from time to time.

> Would be nice to combine walk speed and fly speed to abstract class as it has the same logic.

I was originally considering that, but I wasn't sure of the best place to put the abstract class. Would the same package be fine, or do you have a package/sourceset for abstract and interface code?

eth-p commented:
I couldn't find the formatting errors you mentioned, but I did manage to fix the issues and refactor the handler logic into an abstract class.

I even added a bonus "spectate-speed" flag that takes precedence over the "fly-speed" flag if the player is in spectator mode.

aromaa commented on WG/src/main/java/net/goldtreeservers/worldguardextraflags/flags/Flags.java:
```diff
@@ -32,6 +32,8 @@
 	
 	public final static DoubleFlag WALK_SPEED = new DoubleFlag("walk-speed");
+	public final static DoubleFlag FLY_SPEED = new DoubleFlag("fly-speed");
+	public final static DoubleFlag SPECTATE_SPEED = new DoubleFlag("spectate-speed");
```
I think this is overkill, adds bloat and is really case specific

eth-p replied:
Alright. Will be dropped in my next push.

aromaa commented on WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/AbstractSpeedFlagHandler.java:
```diff
@@ -0,0 +1,90 @@
+public abstract class AbstractSpeedFlagHandler extends HandlerWrapper
+{
+    
+    private Float originalSpeed;
```
Formatting

aromaa commented on WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/AbstractSpeedFlagHandler.java:
```diff
@@ -0,0 +1,90 @@
+	}
+	
+	@Override
+    public void initialize(Player player, Location current, ApplicableRegionSet set)
```
Formatting

aromaa commented on WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/AbstractSpeedFlagHandler.java:
```diff
@@ -0,0 +1,90 @@
+	
+	private void handleValue(Player player, Double speed)
+	{
+        if (speed != null)
```
Formatting

aromaa commented on WG/src/main/java/net/goldtreeservers/worldguardextraflags/wg/handlers/FlySpeedFlagHandler.java:
```diff
@@ -0,0 +1,73 @@
+	
+    public static class Factory extends HandlerWrapper.Factory<FlySpeedFlagHandler>
+    {
+        public Factory(Plugin plugin)
```
Formatting

aromaa reviewed (changes requested):
Marked some formatting errors now also

eth-p commented:
Thanks for pointing out the formatting issues for me. After looking at the other files in the project, it appears that a few of them are either spaces, tabs, or a mix of the two. That would probably explain why my IDE failed to select the correct indentation method 😛

Basing the indentation it off your last commit, I replaced all the spaces from the changed files with tabs.
I also removed the spectator-speed flag as requested.

aromaa reviewed (approved):
Thanks. Looks fine now
